compiler/src/iree/compiler/Codegen/Common/MaterializeUserConfigs.cpp

```diff
@@ -75,16 +110,31 @@ struct MaterializeUserConfigsPass
            << clCodegenTransformDialectLibraryFileName);
     }
 
+    // Run the user specified transform configuration strategy if specified.
+    if (hasTransformConfig) {
```

Comment:
If we combine the two flags, then this logic probably gets simplified.

Reply:
I think the logic got more complicated, but it's partially an artifact of trying to make this kernel config stuff work with the way we used to do transform strategies, which is to use the same strategy everywhere (`__transform_main`). We could get rid of this by requiring that users annotate the hal.executable.exports with `translation_info = None` and never call the IREE TransformDialectInterpreter pass, but I don't feel like breaking all of the various scripts that are floating around right now, nor have to rewrite all the tests. That instability costs me and others time.

Reply:
It did get more complicated. I am actually not sure what the "__kernel_config` thing does now. I thought that `__transform_main` would be enough.

Reply:
We can simplify things at some later point but the gymnastics here are to avoid breaking existing experimental flows with Transform Dialect.

I think the canonical way will be to just have `__transform_main` like you're saying, and if people want the existing functionality, they have to do the annotation with `translation_info = None`. We could even add a one-liner op to do that for them and can simplify all of the logic here at that point. I just need a week or two to find time to do it...

Reply:
I'll file an issue tracking this.